c_char on AIX should be u8

Currently, core::ffi::c_char on AIX is i8, but AFAIK, the default of this platform is unsigned (u8).

[XL C for AIX](https://www.ibm.com/docs/en/xl-c-aix/13.1.3?topic=descriptions-qchars):

> Defaults
> -qchars=unsigned

Clang:

```console
$ clang --version
Homebrew clang version 17.0.6

$ clang -E -dM -x c /dev/null -target powerpc64-ibm-aix | grep __CHAR_        
#define __CHAR_BIT__ 8
#define __CHAR_UNSIGNED__ 1
```

Rust:

```
$ tail -2 src/lib.rs 

const C: core::ffi::c_char = 0u8;

$ cargo check -Zbuild-std=core --target powerpc64-ibm-aix
error[E0308]: mismatched types
    --> src/lib.rs:3972:30
     |
3972 | const C: core::ffi::c_char = 0u8;
     |                              ^^^ expected `i8`, found `u8`
     |
help: change the type of the numeric literal from `u8` to `i8`
     |
3972 | const C: core::ffi::c_char = 0i8;
     |                               ~~
```

cc @ecnelises @bzEq ([target maintainers](https://github.com/rust-lang/rust/blob/548e14b43963882fb758deb89e8258d9b8c2fc2a/src/doc/rustc/src/platform-support/aix.md?plain=1#L7))

### Meta
<!--
If you're using the stable version of the compiler, you should also check if the
bug also exists in the beta or nightly versions.
-->

`rustc --version --verbose`:
```
rustc 1.79.0-nightly (2f090c30d 2024-03-23)
binary: rustc
commit-hash: 2f090c30ddd6b3bbe5c81c087579a5166e7c7278
commit-date: 2024-03-23
host: aarch64-apple-darwin
release: 1.79.0-nightly
LLVM version: 18.1.2
```

--- a/library/core/src/ffi/mod.rs
+++ b/library/core/src/ffi/mod.rs
@@ -133,7 +133,8 @@ mod c_char_definition {
                 any(target_arch = "aarch64", target_arch = "riscv64")
             ),
             all(target_os = "nto", target_arch = "aarch64"),
-            target_os = "horizon"
+            target_os = "horizon",
+            target_os = "aix",
         ))] {
             pub type c_char = u8;
         } else {

